compiler/hash-types/src/new/access.rs

```diff
@@ -1,17 +1,30 @@
 //! Definitions related to access operations.
 
-use super::{symbols::Symbol, terms::TermId};
+use super::{params::ParamTarget, terms::TermId};
 
 /// The kind of an access.
-#[derive(Debug, Copy, Clone, Hash, Eq, PartialEq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum AccessKind {
-    Numeric(usize),
-    Named(Symbol),
+    /// Accessing a constructor field, like `f := Foo(name=3); f.name`.
+    CtorField,
+    /// Accessing a tuple field, like `f := (2, 3); f.0`.
+    TupleField,
+    /// Accessing a module member, like `X := mod { y := 3 }; X.y`.
+    ModMember,
+    /// Accessing a trait member, like `T := trait { y := 3; z := self.y }`
+    TrtMember,
+    /// Accessing a datatype constructor, like `Colour := enum(Red, Green,
+    /// Blue); Colour.Red`
```

Comment:
The access syntax is `Colour::Red`

Reply:
What do you think about keeping `.` for everything? looks cleaner imo

Reply:
Maybe change this at a later stage when the new type-checking implementation is used, right now it's hard to disambiguate between a namespace access and a property access, we've tried it before and it wasn't so easy to make it work with the current implementation.

Reply:
Yeah right now it's not possible. But yeah it should be with the new TC because subjects can always be disambiguated fully (no merges). So for now we will keep `.` and `::`